### PR TITLE
Add websocket token auth in case of different ws domain

### PR DESCRIPTION
**Pull request by darcsoel**
Related issues:
https://github.com/jupyterlab/jupyterlab/issues/9070
https://github.com/jupyterlab/jupyterlab/pull/9080

As user I want to have possibility to use ws on another domain, this PR added one more check to appendToken method.

### CODE REVIEW

File: packages/coreutils/src/url.ts [hunk 1]
```diff
@@ -26,6 +26,9 @@ export namespace URLExt {
     return urlparse(url);
   }
 
+  export function getHostName(url: string): string {
```
Comment (blink1073) [inline packages/coreutils/src/url.ts:29]: Can you please add a docstring for this function?

**Review by blink1073 — APPROVED** [review]
Thanks!  I just had one comment.

File: packages/coreutils/src/url.ts (hunk 1 above)
Comment (darcsoel) [reply packages/coreutils/src/url.ts:36]: Done. Thanks for your comment.

**Review by blink1073 — APPROVED** [review]

**Comment by blink1073** [comment]
Congratulations on your first contribution to JupyterLab @darcsoel!  :tada:

**Pull request merged**